iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object.hpp

```diff
@@ -65,7 +66,7 @@ class SharedMemoryObject : public DesignPattern::Creation<SharedMemoryObject, Sh
                        const uint64_t memorySizeInBytes,
                        const AccessMode accessMode,
                        const OpenMode openMode,
-                       const void* baseAddressHint,
+                       const void* const baseAddressHint,
```

Comment:
I think we use a `nullptr` to indicate that we do not use the hint. What do you think about changing this to `cxx::optional<uint64_t>`?